src/gc/boehm.cr

```diff
@@ -76,16 +76,34 @@ end
 
 # :nodoc:
 fun __crystal_malloc64(size : UInt64) : Void*
+  {% if flag?(:bits32) %}
+    if size > UInt32::MAX
```

Comment:
Thinking about this more generally, is the automatic integer size conversion here a mistake in the language design, because it hides the situations where 64bit integers can be downcoverted to 32bits, and so hides where you should make a check for truncation?

Reply:
Yes, but it's a different issue and of much bigger scope: https://github.com/crystal-lang/crystal/issues/3103

I don't think we can fix this in this same PR. For now we have to do manual checks.

Reply:
I agree, it's far out of scope for this PR.